macosx-x86_64/Makefile

```diff
@@ -11,7 +11,7 @@ target/classes/macosx-x86_64:
 	mkdir -p target/classes/macosx-x86_64
 
 target/%.o : ../libwfssl/src/%.c target/classes/macosx-x86_64
-	$(CC) $(CFLAGS) -Werror -Wall -Wmissing-prototypes -Wstrict-prototypes -Wmissing-declarations -Wpointer-arith -std=c89 -Wdeclaration-after-statement -Wformat -Wformat-security -Wunused -Wno-unknown-pragmas -c $< -o $@ -I../libwfssl/include -I${JAVA_HOME}/include -I${JAVA_HOME}/include/darwin
+	$(CC) $(CFLAGS) -mmacosx-version-min=10.12 -target x86_64-apple-macos10.12 -Werror -Wall -Wmissing-prototypes -Wstrict-prototypes -Wmissing-declarations -Wpointer-arith -std=c89 -Wdeclaration-after-statement -Wformat -Wformat-security -Wunused -Wno-unknown-pragmas -c $< -o $@ -I../libwfssl/include -I${JAVA_HOME}/include -I${JAVA_HOME}/include/darwin
 
 target/classes/macosx-x86_64/libwfssl.dylib: $(OBJ)
-	$(CC) $(CFLAGS) -dynamiclib $(OBJ) -o $@ $(LDFLAGS)
+	$(CC) $(CFLAGS) -mmacosx-version-min=10.12 -target x86_64-apple-macos10.12 -dynamiclib $(OBJ) -o $@ $(LDFLAGS)
```

Comment:
Is there a reason for adding this?

Reply:
10.12 is 5 years old and I thought that this would be already rather old. I was unsure if we want to support older versions than that.

Reply:
If we update this, I think that should be done as a separate issue since this PR is about the new support for aarch64. So I think we can remove this from here for now.

Reply:
Not sure what it WILL generate in that case. Ah yes then it will generate output for the standard architecture which for an M1 machine is aarch64. I changed this by just forcing arch to x86_64 so it generates for that now.